src/appleseed/renderer/kernel/rendering/masterrenderer.cpp

```diff
@@ -354,12 +361,6 @@ struct MasterRenderer::Impl
             return renderer_controller.get_status();
         }
 
-        // Initialize the render device.
-        const bool success =
-            m_render_device->initialize(
-                m_resource_search_paths,
-                m_tile_callback_factory,
-                abort_switch);
         if (!success || abort_switch.is_aborted())
```

Comment:
This piece of code goes together with the block you moved. It needs to be moved as well (and updated since `recorder` won't have been created yet).

Reply:
Hi,
If I am not mistaken this PR can be ignored. 
This problem was further handled by @est77 so I have stopped working on this PR.

Reply:
Also, I just remembered one more thing.
@est77 mentioned that, while solving this issue, he had a problem with Appleseed crushing when rendering would be stopped and restarted (and maybe also when the material was changed).
I am thinking that he forgot about moving this piece of code that you mention here. Maybe that would help.